File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {

        double res=-10000000;
        int n=k;
        int summer=0;
        int l=0;

        for(int r=0;r<nums.length;r++){
            summer+=nums[r];
            if (r==(k-1)){
                res=Math.max(res,summer);
                summer-=nums[l];
                l++;
                k++;
            }
        }
        return res/n;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        l=0
        res=float(-inf)
        sum=0
     
        n=k
        for r in range(len(nums)):
            sum+=nums[r]
            if r==(k-1):
              
                 res=max(res,sum)
              
                
                 sum-=nums[l]
                 l=l+1
                 k=k+1
      
        return res/n
       
            
